utf8.c

```diff
@@ -1685,8 +1685,55 @@ Perl__utf8n_to_uvchr_msgs_helper(const U8 *s,
             *msgs = NULL;
         }
 
+
+/* Returns 0 if no message need to be generated for this problem even if
+ * everything else says to?  Otherwise returns the warning category to use for
+ * the message.
+ *
+ * No message is generated if the UTF8_CHECK_ONLY flag has been set by the
+ * caller.  Otherwise, a message is generated if either:
+ *  1)  the caller has furnished a structure into which messages should be
+ *      returned to it (so it itself can decide what to do); or
+ *  2)  warnings are enabled for either of the the category parameters to the
+ *      macro.  Warnings for the passed-in category are
+ *
+ * The 'warning' parameter is the higher priority warning category to check.
+ * The macro calls ckWARN_d(warning), so warnings for it are considered to be
+ * on by default.
+ *
+ * The second, lower priority category is optional.  To specify not to use one,
+ * call the macro like:   NEED_MESSAGE(WARN_FOO, , 0)'
```

Comment:
Extra `'`.

If you access the parameter as `extra_category + 0`, the caller can omit it entirely because `+ 0` is just zero.

Reply:
thanks